conda/models/index_record.py

```diff
@@ -112,3 +113,11 @@ class IndexRecord(DictSafeMixin, Entity):
     link = ComposableField(Link, required=False)
 
     preferred_env = StringField(default=None, required=False, nullable=True)
+
+    @property
+    def combined_depends(self):
+        from .match_spec import MatchSpec
+        result = {ms.name: ms for ms in (MatchSpec(spec) for spec in self.depends or ())}
+        result.update({ms.name: ms for ms in (MatchSpec(spec, option=True)
+                                              for spec in self.constrains or ())})
+        return tuple(itervalues(result))
```

Comment:
Well, rule number one is that you shouldn't de-dupe if you don't have to, because multiple MatchSpecs with the same name has semantic meaning. So let's make sure you're not doing it in other places. Here, we want the constraints to overwrite the depends. In the absence of duplicates in either list, this is the right way to do it. Do we want to _allow_ dupes in conda recipes? I'd say not. So this should be good. Certainly for now...